src/index.js

```diff
@@ -10,79 +10,87 @@ const {
       urlResolver : new PackageUrlResolver()
     });
 
     analyzer.analyzePackage().then(_package => {
       const metadata = generateAnalysis(_package, '');
-      let elements;
 
-      const processingPackage = !Boolean(
-          metadata.elements.find(item => item.path.indexOf(path.sep) === -1));
+      console.log(`Processing element from the ${args.root} directory`);
 
-      elements = splitByElements(metadata.elements, itemsRootPath);
-
-      console.log(`Processing ${processingPackage ? "package" : "element"} ` +
-                  `from the ${itemsRootPath} directory`);
-
-      createDocFiles(templateFilePath, elements, itemsRootPath, processingPackage);
+      createDocFile(templateFilePath, metadata.elements, args);
     });
   }
 };
 
-function splitByElements(rawElementsData, itemsRootPath) {
-  const elements = {};
+function createDocFile(templateFilePath, elements, args) {
+  fs.readFile(templateFilePath, 'utf-8', (err, data) => {
+    if (err) {
+      throw new Error(`Failed to read the template\n${err.message}`);
+    }
+
+    const docFilePath = `${args.root}${path.sep}index.html`;
+    const docFileContent = fillContent(data, elements, args.root);
 
-  rawElementsData.forEach(item => {
-    const elementRootPath =
-        item.path.substring(0, item.path.indexOf(path.sep)) || getElementName(itemsRootPath);
+    if (!args.overwrite && fs.existsSync(docFilePath)) {
+      const rlInterface = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout
+      });
 
-    if (elements[elementRootPath]) {
-      elements[elementRootPath].push(item);
+      rlInterface.question(`File ${docFilePath} already exists.\nOverwrite it? |yes| `, answer => {
+        fileExistsDialog(answer, rlInterface, () => writeDocFile(docFilePath, docFileContent));
+      });
     } else {
-      elements[elementRootPath] = [item];
+      writeDocFile(docFilePath, docFileContent);
     }
   });
+}
 
-  return elements;
+function fileExistsDialog(answer, rlInterface, successCallback) {
+  const lowerCaseAnswer = answer.toLowerCase();
+  if (answer === "" || lowerCaseAnswer === "yes" || lowerCaseAnswer === "y") {
+    successCallback();
+    rlInterface.close();
+
+  } else if (lowerCaseAnswer === "no" || lowerCaseAnswer === "n") {
+    console.log('Documentation generation canceled.');
+    process.exit(0);
+
+  } else {
+    console.log('Please enter "yes" or "no".\n');
+    rlInterface.question(`Overwrite it? |no| `, answer => {
+      fileExistsDialog(answer, rlInterface, successCallback);
+    });
+  }
 }
 
-function createDocFiles(templateFilePath, elements, itemsRootPath, processingPackage) {
-  fs.readFile(templateFilePath, 'utf-8', (err, data) => {
+function writeDocFile(filePath, content) {
+  fs.writeFile(filePath, content, err => {
     if (err) {
-      throw new Error(`Failed to read the template\n${err.message}`);
+      throw new Error(`Failed to fill the doc file\n${err.message}`);
```

Comment:
Why "fill"? Wouldn't something like "save" or "write" describe what's happening more precisely? Also, I think that it would be more presentable to write "documentation" instead of "doc" (and we're using this term in other messages as well anyway).